src/fmt.rs

```diff
@@ -55,3 +60,118 @@ impl ErrorImpl<()> {
         Ok(())
     }
 }
+
+struct Indented<'a, D> {
+    inner: &'a mut D,
+    ind: Option<usize>,
+    started: bool,
+}
+
+impl<'a, D> Indented<'a, D> {
+    fn numbered(inner: &'a mut D, ind: usize) -> Self {
+        Self {
+            inner,
+            ind: Some(ind),
+            started: false,
+        }
+    }
+
+    fn new(inner: &'a mut D) -> Self {
+        Self {
+            inner,
+            ind: None,
+            started: false,
+        }
+    }
+}
+
+impl<T> fmt::Write for Indented<'_, T>
+where
+    T: fmt::Write,
+{
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        for (ind, mut line) in s.split('\n').enumerate() {
+            if !self.started {
+                // trim first line to ensure it lines up with the number nicely
+                line = line.trim();
+                // Don't render the first line unless its actually got text on it
+                if line.is_empty() {
+                    continue;
+                }
```

Comment:
I am going to remove this logic because starting error messages with <code>"\\n&nbsp;&nbsp;&nbsp;&nbsp;"</code> so that rustc's default `main` prints them on the next line after the "Error:" heading is not a pattern I want to encourage.